MekHQ/src/mekhq/campaign/mission/atb/scenario/HoldTheLineBuiltInScenario.java

```diff
@@ -15,88 +34,91 @@
+
+    @Override
+    public int getScenarioType() {
+        return HOLDTHELINE;
+    }
+
+    @Override
+    public String getScenarioTypeDescription() {
+        return "Hold the Line";
+    }
+
+    @Override
+    public String getResourceKey() {
+        return "holdTheLine";
+    }
+
+    @Override
+    public void setExtraMissionForces(Campaign campaign, ArrayList<Entity> allyEntities,
+            ArrayList<Entity> enemyEntities) {
+        int enemyStart;
+        int playerHome;
+
+        if (isAttacker()) {
+            playerHome = startPos[Compute.randomInt(4)];
+            setStart(playerHome);
+
+            enemyStart = Board.START_CENTER;
+            setEnemyHome(playerHome + 4);
+
+            if (getEnemyHome() > 8) {
+                setEnemyHome(getEnemyHome() - 8);
+            }
+        } else {
+            setStart(Board.START_CENTER);
+            enemyStart = startPos[Compute.randomInt(4)];
+            setEnemyHome(enemyStart);
+            playerHome = getEnemyHome() + 4;
+
+            if (playerHome > 8) {
+                playerHome -= 8;
+            }
+        }
+
+        if (allyEntities.size() > 0) {
+            addBotForce(getAllyBotForce(getContract(campaign), getStart(), playerHome, allyEntities));
+        }
+
+        if (isAttacker()) {
+            addEnemyForce(enemyEntities, getLance(campaign).getWeightClass(campaign), EntityWeightClass.WEIGHT_ASSAULT,
+                    0, 0, campaign);
+        } else {
+            addEnemyForce(enemyEntities, getLance(campaign).getWeightClass(campaign), EntityWeightClass.WEIGHT_ASSAULT,
+                    4, 0, campaign);
+        }
+
+        addBotForce(getEnemyBotForce(getContract(campaign), enemyStart, getEnemyHome(), enemyEntities));
+    }
+
+    @Override
+    public boolean canAddDropShips() {
+        return isAttacker() && (Compute.d6() == 1);
+    }
+
+    @Override
     public void setObjectives(Campaign campaign, AtBContract contract) {
         super.setObjectives(campaign, contract);
-        
+
         ScenarioObjective destroyHostiles = CommonObjectiveFactory.getDestroyEnemies(contract, 33);
-        ScenarioObjective keepFriendliesAlive = CommonObjectiveFactory.getKeepFriendliesAlive(campaign, contract, this, 50, false); 
-        ScenarioObjective keepAttachedUnitsAlive = CommonObjectiveFactory.getKeepAttachedGroundUnitsAlive(contract, this);
-        
-        // attacker must destroy 50%
-        if(!isAttacker()) {
+        ScenarioObjective keepFriendliesAlive = CommonObjectiveFactory.getKeepFriendliesAlive(campaign, contract, this,
+                50, false);
+        ScenarioObjective keepAttachedUnitsAlive = CommonObjectiveFactory.getKeepAttachedGroundUnitsAlive(contract,
+                this);
+
+        // defender must destroy 50%
+        // attacker must destroy 33%
+        if (!isAttacker()) {
             destroyHostiles.setPercentage(50);
             keepFriendliesAlive.setPercentage(33);
```

Comment:
If the attackers must destroy 33%, wouldn't you have to keep 66% alive?

Reply:
Good point.

Reply:
I'm going to address it in a separate PR, as I already made a bunch of commits on top of the branch I used to develop this and it's kind of a pain to separate them out.